dashboard/app/models/user_permission.rb

```diff
@@ -40,4 +42,17 @@ class UserPermission < ActiveRecord::Base
   ].freeze
 
   validates_inclusion_of :permission, in: VALID_PERMISSIONS
+
+  after_save :log_permission_save
+  before_destroy :log_permission_delete
+
+  def log_permission_save
+    HipChat.log "Updating UserPermission: user ID: #{self.user.id}, email: #{self.user.email}, permission: #{self.permission}",
```

Comment:
Both on create and on update: http://apidock.com/rails/ActiveRecord/Callbacks/after_save.

This goes to `infra-production` with current settings. If you want, I can complexify slightly so that it goes to the appropriate room. Thinking about that, I'll change this to noop in the test environment.